src/net/relay.rs

```diff
@@ -1497,6 +1497,11 @@ impl PeerNetwork {
         Ok(())
     }
 
+    // todo(ludo): open question - should nodes be advertising the fact that they just received an attachment?
```

Comment:
Probably not.  Unlike blocks, attachment propagation isn't necessary for producing more attachments, so propagation can be completely pull-based (like it is in Stacks 1.0).

Reply:
The caveat I have in mind is the following: what if the first node receiving and inboxing the attachment is behind a NAT? The attachment would end up being stuck on that machine, no?

Reply:
That's true @lgalabru, but I think we should think of attachments as somewhat less vital than blocks. If, as the registrant of a name, I chose to only broadcast my attachment to a NAT'ed node, it doesn't _really_ impact the network that negatively, it's more of an impact on my own name.